tools/api.py

```diff
@@ -308,18 +311,36 @@ def virtual_param(self, prefix: Optional[str] = None) -> str:
         return self.substitute('${type} ${name}', prefix=prefix)
 
     def virtual_output_args(self, prefix: Optional[str] = None) -> List[str]:
-        return [
-            '&{prefix}{n}'.format(prefix=prefix or '', n=n)
-            for t, n in self.cparams
-        ]
+        container_type = self.type.remove_generic().basic().str()
+        decl_list: List[str] = []
+        for t, n, in self.cparams:
+            if not decl_list and (container_type == "std::vector"
```

Comment:
I would prefer this be done by the `vector_c_wrap` function instead of hardcoding `vector` here.

Reply:
Yes, but I don't find an easier and nicer way to do it generically.